File: src/main/java/net/thedragonteam/armorplus/blocks/ritual/BlockRitualAltar.java
package net.thedragonteam.armorplus.blocks.ritual;

import net.minecraft.block.material.Material;
import net.minecraft.block.state.IBlockState;
import net.minecraft.entity.Entity;
import net.minecraft.entity.item.EntityItem;
import net.minecraft.item.Item;
import net.minecraft.util.math.AxisAlignedBB;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.IBlockAccess;
import net.minecraft.world.World;
import net.thedragonteam.armorplus.blocks.base.BlockBase;
import net.thedragonteam.thedragonlib.util.LogHelper;

import java.util.Objects;
import java.util.Random;

import static net.thedragonteam.armorplus.registry.ModItems.templates;

/**
 * ArmorPlus - Kotlin created by sokratis12GR
 * - TheDragonTeam
 */
public class BlockRitualAltar extends BlockBase {

    public boolean isValid;

    private static final AxisAlignedBB NOT_CONNECTED_AABB = new AxisAlignedBB(0.0625D, 0.0D, 0.0625D, 0.9375D, 0.875D, 0.9375D);

    public BlockRitualAltar() {
        super(Material.ROCK, "ritual_altar", 10F, 10F);
    }

    @Override
    public void onEntityCollidedWithBlock(World worldIn, BlockPos pos, IBlockState state, Entity entityIn) {
        for (Item item : templates)
            if (entityIn instanceof EntityItem && ((EntityItem) entityIn).getItem().getItem() == item) {
                isItemValid(((EntityItem) entityIn).getItem().getItem());
                LogHelper.INSTANCE.info("Is Item Valid: " + isItemValid(((EntityItem) entityIn).getItem().getItem()));
            }
        super.onEntityCollidedWithBlock(worldIn, pos, state, entityIn);
    }

    @Override
    public void updateTick(World worldIn, BlockPos pos, IBlockState state, Random rand) {
        super.updateTick(worldIn, pos, state, rand);
    }

    @Override
    public void onEntityWalk(World worldIn, BlockPos pos, Entity entityIn) {
        for (Item item : templates)
            if (entityIn instanceof EntityItem && ((EntityItem) entityIn).getItem().getItem() == item) {
                isItemValid(((EntityItem) entityIn).getItem().getItem());
                LogHelper.INSTANCE.info("Is Item Valid: " + isItemValid(((EntityItem) entityIn).getItem().getItem()));
            }
        super.onEntityWalk(worldIn, pos, entityIn);
    }

    @SuppressWarnings("deprecation")
    @Override
    public AxisAlignedBB getBoundingBox(IBlockState state, IBlockAccess source, BlockPos pos) {
        return NOT_CONNECTED_AABB;
    }

    private boolean isItemValid(Item item) {
        for (Item valid : templates) {
            isValid = Objects.equals(valid, item);
        }
        return isValid;
    }
}

File: src/main/java/net/thedragonteam/armorplus/blocks/ritual/BlockRitualAltar.kt
/*
 * Copyright (c) TheDragonTeam 2016-2017.
 */

package net.thedragonteam.armorplus.blocks.ritual

import net.minecraft.block.ITileEntityProvider
import net.minecraft.block.material.Material
import net.minecraft.block.state.IBlockState
import net.minecraft.entity.Entity
import net.minecraft.entity.item.EntityItem
import net.minecraft.tileentity.TileEntity
import net.minecraft.util.math.AxisAlignedBB
import net.minecraft.util.math.BlockPos
import net.minecraft.world.IBlockAccess
import net.minecraft.world.World
import net.thedragonteam.armorplus.blocks.base.BlockBase
import net.thedragonteam.armorplus.registry.ModItems
import net.thedragonteam.armorplus.tileentity.TileEntityRitualAltar
import net.thedragonteam.thedragonlib.util.ItemStackUtils.getItemStack
import net.thedragonteam.thedragonlib.util.LogHelper.info

/**
 * ArmorPlus created by sokratis12GR
 * - TheDragonTeam
 */
open class BlockRitualAltar : BlockBase(Material.ROCK, "ritual_altar", 10F, 10F), ITileEntityProvider {

    lateinit private var ritualAltar: TileEntityRitualAltar

    override fun getBoundingBox(state: IBlockState?, source: IBlockAccess?, pos: BlockPos?): AxisAlignedBB {
        return NOT_CONNECTED_AABB
    }

    override fun onEntityWalk(worldIn: World?, pos: BlockPos?, entityIn: Entity?) {
        for (item in ModItems.templates)
            if (entityIn is EntityItem && entityIn.entityItem.item == item) {
                ritualAltar.isItemValid(getItemStack(item))
                info("Is Item Valid: " + ritualAltar.isItemValid(getItemStack(item)))
            }

        super.onEntityWalk(worldIn, pos, entityIn)
    }

    override fun createNewTileEntity(worldIn: World, meta: Int): TileEntity {
        return TileEntityRitualAltar()
    }

    companion object {
        protected val NOT_CONNECTED_AABB = AxisAlignedBB(0.0625, 0.0, 0.0625, 0.9375, 0.875, 0.9375)
    }
}
